source/common/upstream/upstream_impl.cc

```diff
@@ -960,6 +960,44 @@ createOptions(const envoy::config::cluster::v3::Cluster& config,
       config.has_http2_protocol_options(), validation_visitor);
 }
 
+LBPolicyConfig::LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config) {
+
+  switch (config.lb_policy()) {
+    PANIC_ON_PROTO_ENUM_SENTINEL_VALUES;
+  case envoy::config::cluster::v3::Cluster::ROUND_ROBIN:
```

Comment:
Currently, `lb_policy` is not used to set the configs for the lb configs. I don't know why it was done that way (and seems like it could be a source for bugs). 
Could using `lb_policy` to set lb config cause some change in existing behaviour? Do they need to be set independently as is done currently?

Reply:
Talked offline about this a bit and we're not sure if it is enforced that they match. For now I changed them to if statements that check for the config to make it similar to how it behaved before. I opened up an issue [here](https://github.com/envoyproxy/envoy/issues/25583) to discuss it further with the community. If they are forced to match, then I'll come back and change it to a switch.